pilot/pkg/networking/core/v1alpha3/envoyfilter/monitoring.go

```diff
@@ -63,30 +72,39 @@ func IncrementEnvoyFilterMetric(name string, pt PatchType, applied bool) {
 	if !features.EnableEnvoyFilterMetrics {
 		return
 	}
-	result := Applied
-	if !applied {
-		result = Skipped
+	envoyFilterMutex.Lock()
+	defer envoyFilterMutex.Unlock()
+	if _, exists := envoyFilterStatusMap[name]; !exists {
+		envoyFilterStatusMap[name] = make(map[string]bool)
+	}
+	if applied {
+		envoyFilterStatusMap[name][string(pt)] = true
 	}
-	totalEnvoyFilters.With(nameType.Value(name)).With(patchType.Value(string(pt))).
-		With(resultType.Value(string(result))).Record(1)
 }
 
 // IncrementEnvoyFilterErrorMetric increments filter metric for errors.
 func IncrementEnvoyFilterErrorMetric(pt PatchType) {
 	if !features.EnableEnvoyFilterMetrics {
 		return
 	}
-	totalEnvoyFilters.With(patchType.Value(string(pt))).With(resultType.Value(string(Error))).Record(1)
+	envoyFilterStatus.With(patchType.Value(string(pt))).With(resultType.Value(string(Error))).Record(1)
 }
 
-// RecordEnvoyFilterMetric increments the filter metric with the given value.
-func RecordEnvoyFilterMetric(name string, pt PatchType, success bool, value float64) {
-	if !features.EnableEnvoyFilterMetrics {
+func RecordMetrics() {
+	if features.EnableEnvoyFilterMetrics {
 		return
 	}
-	result := Applied
-	if !success {
-		result = Skipped
+	envoyFilterMutex.RLock()
+	defer envoyFilterMutex.RUnlock()
+	for name, pmap := range envoyFilterStatusMap {
+		for pt, applied := range pmap {
+			result := Applied
+			if !applied {
+				result = Skipped
+			}
+			envoyFilterStatus.With(nameType.Value(name)).With(patchType.Value(pt)).
```

Comment:
I still don't see where we ever record 0 if we remove it?

Reply:
I thought setting Skipped with 1 covers that case. But it does not make sense - I will set it to 0 with Applied result.